### PR TITLE
lfortran: update to 0.34.0

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by pmetzger** [comment]
What is the reason for disabling tests?

**Comment by barracuda156** [comment]
> What is the reason for disabling tests?

Because CI are broken and `test.run no` has no effect here. I tried that.

(And then, CI fail on a silly permission error, but even if that is fixed, one of test cases gonna fail.)

File: lang/lfortran/Portfile
Comment (reneeotten) [inline lang/lfortran/Portfile:52]: I've asked this probably last update as well - does it work with Python 3.12? If so, please switch.

File: lang/lfortran/Portfile
Comment (reneeotten) [inline lang/lfortran/Portfile:105]: why does this only appear to be a problem for some of your ports? The CI works perfectly fine for 99% of the cases...

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: lang/lfortran/Portfile
Comment (barracuda156) [reply lang/lfortran/Portfile:105]: Well, we know why was that the issue with OCaml and R ports, but I have no idea why it happens here, since the port does not use custom PG overriding test phase.

File: lang/lfortran/Portfile
Comment (barracuda156) [reply lang/lfortran/Portfile:52]: It will be done once I fix it.

Alternatively, of course, Macports may remove older Pythons, and just make one single version to be used by every port.

P. S. It is generally a bad practice to switch something like this just for the sake of fashion. And so far I have not seen a single demonstrable case why we need Python 3.12 here or in any other port of mine. Obviously, we cannot use Python 3.7, since that one is broken on some newer systems, but AFAIK Python 3.9+ work fine.

File: lang/lfortran/Portfile
Comment (kencu) [reply lang/lfortran/Portfile:105]: @jmroot -- a ruling please.

@barracuda156 will not update any of his ports to use MacPorts currently designated python 3.12, because python 3.12 doesn't work on 10.6 PPC.

@reneeotten  and others keep asking him to do so, which is getting tedious.

What would you like done?

Comment (barracuda156) [reply lang/lfortran/Portfile:105]: @kencu If you or anyone with commit access want to modify my ports, I have no way to prevent that. But indeed, don’t make me involved, please. I am not gonna break my own ports for no good reason.
Somebody else can break them, but no one can force me to do that. This is not my employment, after all, I have no such an obligation.

If a change _fixes nothing_ but _breaks something,_ it should not be done. Especially without testing. Simple as that. Even if it breaks a system with three users. It still improves nothing.
Does `lpython` upstream say that Python 3.11 cannot work with it? That will be a credible reason to switch it, of course. But not anyone’s personal preferences, sorry.

Instead of wasting time to force switch a working Python for a less working one, maybe you could fix a broken GCC in Macports? What is now Macports’s default (13.2.0) was made such without proper testing, and it does not work with `libc++` (yeah, 10.6 ppc and Leopard are fine, while Sonoma is broken).
This is by the way an illustration why blind updates of this sort can be harmful. But no one cares, this is broken for months. Much more important is to change default Python, and specifically in my ports, and not hundreds of others which also use older versions.

Comment (kencu) [reply lang/lfortran/Portfile:105]: Of course, I don't want to modify anyone's ports, and I don't want to have to talk about this issue at all ever again! 

So let's just get a ruling, and have this over with, whatever it is, so people stop having friction about this one thing. Josh may well say you can do what you want with your 3000+ ports and keep them all at python 3.11, etc. But we don't want to keep bring this up every single PR like it has been.

Comment (barracuda156) [reply lang/lfortran/Portfile:105]: @kencu By the way, if you had in mind to make a genuine improvement, here is one way to do it, satisfying all sides, perhaps:

Have an umbrella port for Python to run scripts (it is just build system and/or running tests, any Python would do), and then you can set whatever you prefer for Intel and arm64, and I can have my powerpc-only port, which is used only on PowerPC systems. Everyone is happy then?

Comment (kencu) [reply lang/lfortran/Portfile:105]: Josh set a policy that all ports use MacPorts designated python (currently 3.12) unless the port cannot use it.

I didn't set the policy (although I agree with it).

Don't take it so personally -- if it is the policy, it's the policy. Abide by it.

**Comment by jmroot** [comment]
> > What is the reason for disabling tests?
> 
> Because CI are broken and `test.run no` has no effect here. I tried that.
> 
> (And then, CI fail on a silly permission error, but even if that is fixed, one of test cases gonna fail.)

Yeah don't do that. CI never runs the portfile's tests currently. If the built-in arch test is failing due to unreadable files, set `test.ignore_archs yes` until 2.9.2 is released.

File: lang/lfortran/Portfile
Comment (jmroot) [reply lang/lfortran/Portfile:105]: The policy is phrased as a "should" not a "must", but yes, please don't hold newer systems hostage to the needs of old ones. In this case it should not be hard to use 3.12 when it works and a different version when it doesn't.

Comment (ryandesign) [reply lang/lfortran/Portfile:105]: I too stopped blindly updating ports to use python 3.12 because we already had a couple instances where doing so broke ports (e.g. https://trac.macports.org/ticket/69036 and https://trac.macports.org/ticket/69228) and I am not a python expert and am not interested in investigating how to fix those breakages. I leave it to someone else to do in many cases.

> CI should be fixed to respect `test.run no`

MacPorts base was [enhanced in 2.9.0](https://github.com/macports/macports-base/commit/ddc26554e6ed49ce8501069b967284739fa6d34e) to have a built-in test method that is used for all ports, even those that do not set `test.run yes`. CI was [deliberately changed](https://github.com/macports/macports-ports/commit/d3219b762afe61e965be308d1a165f35c944abbd) to run those built-in tests. There is a bug in MacPorts base that causes the built-in tests to fail if a file is unreadable. This will be [fixed in MacPorts 2.9.2](https://github.com/macports/macports-base/commit/e3468c5d60b9a5872d37b57c3d0e6a60694daac9).

Comment (kencu) [reply lang/lfortran/Portfile:105]: OK, so I would take that to mean that the PR reviewers who keep pointing out to @barracuda156 that he is meant to use python 3.12 can stop doing so.

That should improve everyone’s interactions nicely, and we can move on back to the fun stuff.

Comment (reneeotten) [reply lang/lfortran/Portfile:105]: to me it read as "yes, use Python 3.12 wherever it works and if you really want to restrict old systems to an earlier version". But I will just stop asking and/or interacting with PRs like this and leave it up to whoever wants to merge things. 

I agree of course that blindly updating to Python 3.12 should not be done, but I'm pretty sure the majority of the ports will work just fine with Python 3.12.

Comment (barracuda156) [reply lang/lfortran/Portfile:105]: @jmroot Thank you, Josh, noted.

@reneeotten As you well know, I do not avoid python 3.12 everywhere out of principle, and normally pro-actively add it to Python ports. But there is really no urgency to rush a switch for non-Python ports which use Python for technical needs, as long as existing choice works for all supported systems.

**Comment by barracuda156** [comment]
> > > What is the reason for disabling tests?
> > 
> > 
> > Because CI are broken and `test.run no` has no effect here. I tried that.
> > (And then, CI fail on a silly permission error, but even if that is fixed, one of test cases gonna fail.)
> 
> Yeah don't do that. CI never runs the portfile's tests currently. If the built-in arch test is failing due to unreadable files, set `test.ignore_archs yes` until 2.9.2 is released.

Awesome, I missed that option.

File: lang/lfortran/Portfile
Comment (kencu) [reply lang/lfortran/Portfile]: That is not really true, though.

One of the largest complaints agains MacPorts as compared to homebrew is bloat, and many people find they need to have multiple versions of perl, python, llvm, clang, and other things installed.

People leave MacPorts for this exact reason.

So to address those complaints, MacPorts agreed to standardize on one python version (and perl version) to be used above all others, and only use another version if there was some good reason to do so (like the recommended version doesn't work).

So by your forcing everyone to keep python3.11 around to suit your 10.6-PPC purposes, you do indeed cause needless bloat and worsen (and undermine) that process.

Comment (kencu) [reply lang/lfortran/Portfile]: However, the admins don't want to be too dictatorial or chase you away, so they acquiesce to this, solely and uniquely for you, pretty much.

Comment (barracuda156) [reply lang/lfortran/Portfile]: @kencu 
> One of the largest complaints agains MacPorts as compared to homebrew is bloat, and many people find they need to have multiple versions of perl, python, llvm, clang, and other things installed.

_This is very well beyond the scope of present discussion._
I see advantages and disadvantages with either of approaches, but I am pretty sure that the market does not need another identical product.
Perhaps with modern hardware and internet speed (and pre-built ports) the problem is more imaginary than practical. For me on a PowerMac it is a pain that Macports needs to build 4–5 empty libgcc ports which takes a full day with strictly none positive utility, but with pre-built ports on arm64 we have much more versatility, power and compatibility at the cost of few extra minutes of downloading and occasionally compiling something.

> So by your forcing everyone to keep python3.11 around to suit your 10.6-PPC purposes, you do indeed cause needless bloat and worsen (and undermine) that process.

At least at the moment there are numerous ports defaulting to 3.11 and even earlier, so I don’t think right now this increases havoc.

I get everyone’s points, let’s really save our time for a more productive topic, which we actually have. (And I hope we find a neat solution for Pythons too.)
We will not keep it stuck at 3.11 indefinitely in any case.

Comment (kencu) [reply lang/lfortran/Portfile]: I don't think you do get everyone's points, and this is a very productive topic -- needlessly forcing everyone to stay on python 3.11 solely so you can make a point about getting your way supporting 10.6 for PPC is very selfish.

**Pull request merged**
